internal/migrations/20251217120000_convert_network_columns_to_native_types.go

```diff
@@ -0,0 +1,176 @@
+		err := tx.Exec(fmt.Sprintf(`
+			DELETE FROM %s WHERE %s IS NULL OR %s = ''
+		`, table, column, column)).Error
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete empty/null rows in %s.%s", table, column)
+		}
+
+		// Check for values that don't look like valid IPs or CIDRs
+		// Valid patterns: IPv4, IPv6, IPv4/prefix, IPv6/prefix
+		// This regex is permissive - it catches obvious garbage but lets PostgreSQL do final validation
+		var invalidValues []string
+		err = tx.Raw(fmt.Sprintf(`
+			SELECT DISTINCT %s FROM %s 
+			WHERE %s !~ '^[0-9a-fA-F.:]+(/[0-9]+)?$'
+			LIMIT 10
+		`, column, table, column)).Scan(&invalidValues).Error
+		if err != nil {
+			return errors.Wrapf(err, "failed to validate %s.%s", table, column)
+		}
+
+		if len(invalidValues) > 0 {
+			return fmt.Errorf("found invalid network values in %s.%s: %v (showing up to 10)", table, column, invalidValues)
+		}
+
+		// For CIDR columns, verify they have a prefix (e.g., /24)
+		if isCidr {
+			var missingPrefix []string
+			err = tx.Raw(fmt.Sprintf(`
+				SELECT DISTINCT %s FROM %s 
+				WHERE %s !~ '/[0-9]+$'
+				LIMIT 10
+			`, column, table, column)).Scan(&missingPrefix).Error
+			if err != nil {
+				return errors.Wrapf(err, "failed to check CIDR prefix in %s.%s", table, column)
+			}
+
+			if len(missingPrefix) > 0 {
+				return fmt.Errorf("found CIDR values without prefix in %s.%s: %v (showing up to 10). CIDRs must include prefix like /24", table, column, missingPrefix)
+			}
+		}
+
+		// Try a test conversion to inet to catch values that fail parsing
+		var badCount int64
+		err = tx.Raw(fmt.Sprintf(`
+			SELECT COUNT(*) FROM %s 
+			WHERE NOT (
+				%s::text ~ '^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+(/[0-9]+)?$'
+				OR %s::text ~ '^[0-9a-fA-F:]+(/[0-9]+)?$'
+			)
+		`, table, column, column)).Scan(&badCount).Error
+		if err != nil {
+			// If the query itself fails, the data format might be really bad
+			return errors.Wrapf(err, "failed to validate format of %s.%s", table, column)
+		}
+
+		return nil
+	}
+
+	migrate := func(tx *gorm.DB) error {
+		for _, col := range columns {
+			// Check current column type
+			var dataType string
+			err := tx.Raw(`
+				SELECT data_type FROM information_schema.columns 
+				WHERE table_name = ? AND column_name = ?
+			`, col.table, col.column).Scan(&dataType).Error
+			if err != nil {
+				return err
+			}
+
+			// Only convert if column is text/varchar (not already inet/cidr)
+			if dataType != "text" && dataType != "character varying" {
+				continue
+			}
+
+			// Clean up empty/null rows and validate remaining data before conversion
+			if err := cleanAndValidateNetworkData(tx, col.table, col.column, col.isCidr); err != nil {
+				return errors.Wrapf(err, "data cleanup/validation failed for %s.%s", col.table, col.column)
+			}
+
+			// Convert to native type
+			// For CIDR columns: use network(column::inet) to normalize host bits to zero
+			// For IP columns: use direct cast to inet
+			var usingClause string
```

Comment:
nit: in general else condition is just making the code harder to read. This could be:
```
// defaults to IP column behaviour
usingClause = fmt.Sprintf("%::%s",  col.column, col.targetType)
if col.isCidr {
    // network() extracts the network address from an inet value, returning a cidr
	// This normalizes values like "192.168.1.5/24" to "192.168.1.0/24"
    usingClause = fmt.Sprintf("network(%s::inet)", col.column)
}
```